salt/modules/file.py

```diff
@@ -3174,17 +3174,37 @@ def patch(originalfile, patchfile, options="", dry_run=False):
     options
         Options to pass to patch.
 
+    .. note::
```

Comment:
Is it not the case that any distribution of `patch` on the PATH should work?
A standalone Windows port or a mingw build or a WSL version?

Reply:
Sure all install variations of `patch` should be fine, and this comment isn't meant to restrict users to only using that installation method but rather this comment brings the changes inline with our golden images, as that's the installation method chosen

Reply:
+ i haven't manually verified that all the other means of obtaining patch on a windows system, just this route

Reply:
It reads like you have to install git-for-windows in order to use it

Reply:
Alright that's fair, how about I update to say this instead:
```
Windows now supports using patch as of 3004. In order to use this function in Windows, 
please install the patch binary through your own means and ensure it's found in the system Path.
If installing through git-for-windows, please select the optional
"Use Git and optional Unix tools from the Command Prompt" option when installing Git.
```

With this change, it's also appropriate to remove this
```
        if salt.utils.platform.is_windows():
            raise CommandExecutionError(
                "Please check your git-for-windows installation "
                "to ensure Git\\usr\\bin\\ is found in your "
                "system Path variable."
            )
```
and instead let the original raise CommandExecutionError be left in place to simplify.